activesupport/lib/active_support/deprecation/proxy_wrappers.rb

```diff
@@ -145,6 +145,10 @@ def inspect
         target.inspect
       end
 
+      # Don't give a deprecation warning on methods that IRB may invoke
+      # during tab-completion.
+      delegate :hash, :instance_methods, to: :target
```

Comment:
I considered future-proofing this with
```ruby
delegate *BasicObject.public_methods, to: :target
```
but that seems too heavy-handed.  Would
```ruby
delegate :hash, *BasicObject.public_methods.grep(/methods/), to: :target
```
be better?